### DISCUSSION TITLE
Pass flags into FFMpeg for low latency

### DISCUSSION THREAD

**Discussion by Videstra** [unosquare/ffmediaelement#601] [Q&A] [answered]
Hi: 
I am using this great control to decode an rtsp stream. Is there a way to pass flags into ffmpeg through the mediaelement?  For my application I require very low latency which I can get from ffmpeg (ffplay) by passing **-fflags nobuffer -flags low_delay -framedrop** 
Our application allows panning/zooming so actual video quality is of less importance than the ability to see the result quickly.  The above flags do a very nice job with straight ffplay.
If anyone has advice on how to set up the control for low latency I would appreciate it very much!

**Comment by Videstra** [comment, marked as answer]
Well - turns out the method is included in the sample code provided! DOH! Although it doesn't appear to actually do the same thing that happens when I pass the flags to ffplay - but that is likely due to different versions of the dlls (vs the static exe I tested the flags with).
The method is pretty simple though:  In VB: 
```
    Private Sub OnMediaInitializing(sender As Object, e As Unosquare.FFME.Common.MediaInitializingEventArgs) Handles Media.MediaInitializing
        e.Configuration.PrivateOptions("rtsp_trasport") = "tcp"  '<-this may not help latency as UDP will be quicker, but I tried both
        e.Configuration.PrivateOptions("flags") = "low_delay"
        e.Configuration.GlobalOptions.FlagNoBuffer = True
        e.Configuration.GlobalOptions.MaxAnalyzeDuration = TimeSpan.Zero
    End Sub

```
If anyone has any suggestions for effectively reducing latency, I would be most appreciative!   The latency is _usable_, but not great.

**Reply by Wolfleader101** [reply]
This doesn't seem to reduce any lag for me...

**Reply by Videstra** [reply]
Correct.  I just entered a new issue.  #606 
Although I can set low latency flags with **ffplay** (using the same dlls I am using in the **ffmediaelement**) those same flags do not appear to do anything in the **ffmediaelement**.  The C# sample shows the flags in **PrivateOptions** without the little dash, I have tried both ways and still no lowered latency.

**Reply by Wolfleader101** [reply]
> Correct. I just entered a new issue. #606 Although I can set low latency flags with **ffplay** (using the same dlls I am using in the **ffmediaelement**) those same flags do not appear to do anything in the **ffmediaelement**. The C# sample shows the flags in **PrivateOptions** without the little dash, I have tried both ways and still no lowered latency.


Ye, I've also made an issue haha #605 - For some reason, it just doesn't want to pass the parameters......


However I was able to reduce the delay slightly as mentioned in my latest comment in #605
